pkg/database/database.go

```diff
@@ -0,0 +1,57 @@
+// Copyright KubeArchive Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package database
+
+import (
+	"database/sql"
+	"fmt"
+	"github.com/kubearchive/kubearchive/pkg/models"
+	_ "github.com/lib/pq"
+)
+
+type DBInterface interface {
+	QueryResources(kind, group, version string) ([]models.Resource, error)
+}
+
+type Database struct {
+	db                *sql.DB
+	resourceTableName string
+}
+
+func NewDatabase() (*Database, error) {
+	dataSource, err := ConnectionStr()
+	if err != nil {
+		return nil, err
+	}
+	db, err := sql.Open("postgres", dataSource)
+	if err != nil {
+		return nil, err
+	}
+	if err = db.Ping(); err != nil {
+		return nil, err
+	}
+	return &Database{db, "archive"}, nil
+}
+
+func (db *Database) QueryResources(kind, group, version string) ([]models.Resource, error) {
+	query := fmt.Sprintf( //nolint:gosec
+		"SELECT data FROM %s WHERE kind=$1 AND api_version=$2", db.resourceTableName) //nolint:gosec
```

Comment:
I think that format string should be made a constant that is declared outside of this function so it will be easier to find and modify if we need to in the future.

Also instead of using `Sprintf` to insert the values being looked for, i think we should use [sql.NamedArg instead](https://pkg.go.dev/database/sql#NamedArg)

Reply:
I tried to use an approach using the sql.NamedArg (with that I could remove the `//nolint:gosec` comment too) but I didn't find a way to do it for the name of the tables.

If I'm not mistaken, based  on [this post in StackOverflow](https://stackoverflow.com/questions/65327759/sql-table-name-as-variable-to-query) I think that there is no way to provide the table name as a variable without doing a string substitution, but please feel free to correct me if there is something I missunderstood or overlooked.

For extracting the SQL query, I think that is a good idea, and I did the change :)

Reply:
I didn't realize that you couldn't use NamedArg for the table name